core/meta/src/TStreamerElement.cxx

```diff
@@ -480,8 +480,10 @@ void TStreamerElement::ls(Option_t *) const
 void TStreamerElement::SetArrayDim(Int_t dim)
 {
    fArrayDim = dim;
-   if (dim) fType += TVirtualStreamerInfo::kOffsetL;
-   fNewType = fType;
+   if (dim) {
+      fType += TVirtualStreamerInfo::kOffsetL;
+      fNewType += TVirtualStreamerInfo::kOffsetL;
+   }
```

Comment:
It is a precursor to other commits (in this PR and the next) move code to rely more on `fNewType` for the case of the `TStreamerElement` describing the current/in-memory schema.    If I remember correctly, 'just' changing this does not affect any of the code path **prior** to this PR.  As we extend the support for write time transformation it would become more important (likely seen in an example storing a fixed size  array of enums.

As it is, it was a logical/obvious error.

Reply:
And I think there might be a array of enums somewhere in roottest.  Either way, I will extend the enum test to include that case.